src/spec.c

```diff
@@ -62,7 +64,11 @@ IndexSpec *IndexSpec_ParseRedisArgs(RedisModuleCtx *ctx, RedisModuleString *name
     args[i] = RedisModule_StringPtrLen(argv[i], NULL);
   }
 
-  return IndexSpec_Parse(RedisModule_StringPtrLen(name, NULL), args, argc, err);
+  IndexSpec *ret = IndexSpec_Parse(RedisModule_StringPtrLen(name, NULL), args, argc, err);
+  if (ret) {
+    IndexSpec_StartGC(ctx, ret, GC_DEFAULT_HZ);
```

Comment:
This looks really dirty. Maybe you should abstract this when a new IndexSpec is created?